libs/librepcb/common/graphics/graphicsview.cpp

```diff
@@ -203,6 +203,39 @@ void GraphicsView::zoomAnimationValueChanged(const QVariant& value) noexcept {
  *  Inherited from QGraphicsView
  ******************************************************************************/
 
+bool GraphicsView::event(QEvent* event) {
+    if (event->type() == QEvent::NativeGesture) {
+        QNativeGestureEvent* gesture = dynamic_cast<QNativeGestureEvent*>(event);
+        switch (gesture->gestureType()) {
+          case Qt::BeginNativeGesture:
+          case Qt::EndNativeGesture: {
+            return true;
+          }
+          case Qt::ZoomNativeGesture: {
+            QTransform transform;
+            transform.scale(1+gesture->value(), 1+gesture->value());
+            setTransform(transform, true);
+            return true;
+          }
+          default: {
+            // Only process zoom gestures
+            return false;
+          }
+        }
+    }
+
+    return false;
+}
+
+// This function receives the wheel events for the viewport widget.
+void GraphicsView::wheelEvent(QWheelEvent* event) {
+  if (event->source() == Qt::MouseEventSynthesizedBySystem) {
```

Comment:
The `source()` method is not available in Qt < 5.5, but this can easily be fixed by adding a #if/endif:
```cpp
#if (QT_VERSION >= QT_VERSION_CHECK(5, 5, 0))
  // Qt >= 5.5
#else
  // Qt < 5.5
#endif
```

Reply:
Done.